src/main/java/no/sikt/nva/BrageProcessor.java

```diff
@@ -65,20 +82,19 @@ private List<Record> processBundles(List<File> resourceDirectories) {
 
     private Optional<Record> processBundle(LicenseScraper licenseScraper,
                                            File entryDirectory) {
-        var record = new Record();
-        record.setOrigin(Path.of(destinationDirectory, entryDirectory.getName()));
+        var brageLocation = new BrageLocation(Path.of(destinationDirectory, entryDirectory.getName()));
         try {
-            var bundlePath = entryDirectory.getPath();
-            var handlePath = Path.of(bundlePath, HANDLE_DEFAULT_NAME);
-            var dublinCoreFile = new File(entryDirectory, DUBLIN_CORE_XML_DEFAULT_NAME);
-            var dublinCore = DublinCoreFactory.createDublinCoreFromXml(dublinCoreFile, record.getOriginInformation());
-            record.setId(HandleScraper.extractHandleFromBundle(handlePath, dublinCoreFile));
-            DublinCoreParser.validateAndParseDublinCore(dublinCore, record);
+            var dublinCore = DublinCoreFactory.createDublinCoreFromXml(getDublinCoreFile(entryDirectory));
+            brageLocation.setTitle(DublinCoreParser.extractTitle(dublinCore));
+            brageLocation.setHandle(
+                handleScraper.scrapeHandleFromRescueMapOrHandleFileOrDublinCore(getHandlePath(entryDirectory),
+                                                                                dublinCore));
```

Comment:
Det virker som at det blir en generell metode for scraping av handle, så kanskje rename den til srapeHandle?

Reply:
fikset